keps/sig-cli/4292-kubectl-debug-custom-profile/README.md

```diff
@@ -0,0 +1,981 @@
+
+Be sure to consider highly-available clusters, where, for example,
+feature flags will be enabled on some API servers and not others during the
+rollout. Similarly, consider large clusters and how enablement/disablement
+will rollout across nodes.
+-->
+
+User may pass a configmap(or any other that is used by other components) in custom profiling
+and during debugging, user may modify (intentionally or unintentionally) these resources in an unexpected way for
+the workloads. But this scenario is not different from any simple pod creation. Since there
+is no scenario impacting the workloads specific to this feature. My answer is no to this question.
+
+###### What specific metrics should inform a rollback?
+
+<!--
+What signals should users be paying attention to when the feature is young
+that might indicate a serious problem?
+-->
+NA
+
+###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
+
+<!--
+Describe manual testing that was done and the outcomes.
+Longer term, we may want to require automated upgrade/rollback tests, but we
+are missing a bunch of machinery and tooling and can't do that now.
+-->
+NA
+
+###### Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?
+
+<!--
+Even if applying deprecation policies, they may still surprise some users.
+-->
+NA
+
+### Monitoring Requirements
+
+<!--
+This section must be completed when targeting beta to a release.
+
+For GA, this section is required: approvers should be able to confirm the
+previous answers based on experience in the field.
+-->
+NA
+
+###### How can an operator determine if the feature is in use by workloads?
+
+<!--
+Ideally, this should be a metric. Operations against the Kubernetes API (e.g.,
+checking if there are objects with field X set) may be a last resort. Avoid
+logs or events for this purpose.
+-->
+Users can determine by checking the value of `KUBECTL_DEBUG_CUSTOM_PROFILE` environment variable.
+However, from the cluster admin point of view, all the APIs that this feature use have already been GA-ed,
+so that it is hard to distinguish whether users enable this feature on their locals after exporting
+the feature environment variable.
+
+###### How can someone using this feature know that it is working for their instance?
+
+<!--
+For instance, if this is a pod-related feature, it should be possible to determine if the feature is functioning properly
+for each individual pod.
+Pick one more of these and delete the rest.
+Please describe all items visible to end users below with sufficient detail so that they can verify correct enablement
+and operation of this feature.
+Recall that end users cannot usually observe component logs or access metrics.
+-->
+
+- [ ] Events
+  - Event Reason: 
+- [ ] API .status
+  - Condition name: 
+  - Other field: 
+- [X] Other (treat as last resort)
+  - Details: Checking the value of `KUBECTL_DEBUG_CUSTOM_PROFILE` environment variable
```

Comment:
They can run a test debug and see if their profile is respected in the resulting container?

Reply:
Thanks for review @BenTheElder. I liked your comment above ^ and have added it to this section.